modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/filters/modal_content/selection_filter/SelectionFilter.tsx

```diff
@@ -84,272 +233,607 @@ function Body({
+													selectedPicklist: item,
+													sourceType,
+												})
+											);
+										}}
 									/>
-								</ClayRadioGroup>
-							</ClayForm.Group>
-							<ClayForm.Group
-								className={classNames({
-									'has-error': !isValidSingleMode,
-								})}
-							>
-								<label htmlFor={preselectedValuesFormElementId}>
-									{Liferay.Language.get('preselected-values')}
-
-									<span
-										className="label-icon lfr-portal-tooltip ml-2"
-										title={Liferay.Language.get(
-											'choose-values-to-preselect-for-your-filters-source-option'
-										)}
-									>
-										<ClayIcon symbol="question-circle-full" />
-									</span>
-								</label>
-
-								<CheckboxMultiSelect
-									allowsCustomLabel={false}
-									aria-label={Liferay.Language.get(
-										'preselected-values'
-									)}
-									inputName={preselectedValuesFormElementId}
-									items={preselectedValues.map((item) => ({
-										label: item.name,
-										value: String(
-											item.externalReferenceCode
-										),
-									}))}
-									loadingState={4}
-									onChange={setPreselectedValueInput}
-									onItemsChange={(selectedItems: any) =>
-										onPreselectedValuesChange(
-											selectedItems.map(
-												({value}: any) => {
-													return selectedPicklist.listTypeEntries.find(
-														(item) =>
-															String(
-																item.externalReferenceCode
-															) === String(value)
+								)}
+
+								{selectedPicklist && (
+									<>
+										<ClayLayout.SheetSection className="mb-4">
+											<h3 className="sheet-subtitle">
+												{Liferay.Language.get(
+													' filter-options'
+												)}
+											</h3>
+										</ClayLayout.SheetSection>
+										<ClayForm.Group
+											className={classNames({
+												'has-error': !isValidSingleMode,
+											})}
+										>
+											<label
+												htmlFor={
+													preselectedValuesFormElementId
+												}
+											>
+												{Liferay.Language.get(
+													'preselected-values'
+												)}
+
+												<span
+													className="label-icon lfr-portal-tooltip ml-2"
+													title={Liferay.Language.get(
+														'choose-values-to-preselect-for-your-filters-source-option'
+													)}
+												>
+													<ClayIcon symbol="question-circle-full" />
+												</span>
+											</label>
+
+											<CheckboxMultiSelect
+												allowsCustomLabel={false}
+												aria-label={Liferay.Language.get(
+													'preselected-values'
+												)}
+												inputName={
+													preselectedValuesFormElementId
+												}
+												items={preselectedValues.map(
+													(item) => ({
+														label: item.name,
+														value: String(
+															item.externalReferenceCode
+														),
+													})
+												)}
+												loadingState={4}
```

Comment:
What does this 4 mean?

Reply:
Not sure...it was already there. In Clay docs are used as the regular value and behavior looks fine, maybe Clay team could answer to it